Unread Count inconsistencies

mahendrakalkura opened the pull request:
1. When a user presses `m`:

- Unread Count in the `titlebar` is updated
- Unread Count in the `menu` is updated
- Unread Count in the `header` is updated

There is no issue here.

2. When a user presses `o`:

- Unread Count in the `menu` is not updated

This issue has been fixed.

3. When a user presses `v`:

- Unread Count in the `titlebar` is not updated
- Unread Count in the `menu` is not updated
- Unread Count in the `header` is not updated

These issues have been fixed.

fguillot commented on ui/static/js/app.js:
```diff
@@ -293,6 +293,11 @@ class UnreadCounterHandler {
+        // is updated correctly when users presses "v".
+        let oldValue = parseInt(document.title.split('(')[1], 10);
+        let newValue = callback(oldValue);
+        document.title = 'Unread Items (' + newValue + ') - Miniflux';
```
There is a problem here. You should not hard code the title in English because the application internationalized. In addition to that, if you are on another "items list" page which is not the unread page, the wrong title get replaced.

mahendrakalkura commented:
> Please, commit only the necessary files.

OK. In this case, it will be only `ui/static/js/app.js`. I will update the PR shortly.

> if you are on another "items list" page which is not the unread page, the wrong title get replaced.

Good catch. I did not see this (may be because I never used anything but the unread page). :)

I will fix this like so and update the PR shortly.

```
if (window.location.href ends with "/unread") { // pseudocode
    document.title = ...
}
```

> You should not hard code the title in English because the application is internationalized. 

I realized this pretty early on, however, I do not know how to do this on a per-page basis. As I see now, locale-specific strings are only available in the Go layer.

I had the following idea in mind.

```
<body 
    data-title-unread="..."
    data-title-history="..."
    ...
    >

if (window.location.href ends with "/unread") {
    document.title = document.body.dataset.titleUnread // replace unread count placeholder with the actual number, 
                                                       // if greater than 0
}
if (window.location.href ends with "/history") {
    document.title = document.body.dataset.titleUnread // replace count placeholder with the actual number
}
...
```

Is this acceptable?

mahendrakalkura commented:
> Please, commit only the necessary files.

Done.

> You should not hard code the title in English because the
> application is internationalized.

I have fixed this using a simpler/shorter method. Let me know if this is acceptable.

https://github.com/miniflux/miniflux/pull/34/files#diff-b90d0484664762feb471bb4dd1a6e88bR324

> if you are on another "items list" page which is not the unread
> page, the wrong title get replaced.

I have fixed this.

https://github.com/miniflux/miniflux/pull/34/files#diff-b90d0484664762feb471bb4dd1a6e88bR311

mahendrakalkura commented:
@fguillot Let me know if you need any other changes from me.

fguillot commented on ui/static/js/app.js:
```diff
@@ -2,6 +2,16 @@
 'use strict';
 
+// Polyfill for Internet Explorer
+if (!String.prototype.endsWith) {
```
Miniflux supports only ES6 browsers, you can remove this polyfill.

mahendrakalkura replied:
Removed.

fguillot commented on ui/static/js/app.js:
```diff
@@ -322,11 +360,18 @@ class EntryHandler {
+                // updated consistently across the following callsites.
+                // - EntryHandler.updateEntriesStatus()
+                // - EntryHandler.markEntryAsRead
+                // - NavHandler.markPageAsRead
```
If the code is not necessary anymore, just remove it instead of adding comments.

mahendrakalkura replied:
Removed.

mahendrakalkura commented:
> Make sure to rebase your changes with the latest master. 

Done.

> I will test your PR this week.

OK.

> Miniflux supports only ES6 browsers, you can remove this polyfill.

Removed.

> If the code is not necessary anymore, just remove it instead of adding comments.

Removed.

mahendrakalkura commented:
@fguillot Thanks.
